### PR TITLE
Price Floors: Imp level floorMin

### PR DESCRIPTION
**Pull request by SerhiiNahornyi**

### CODE REVIEW

File: src/main/java/org/prebid/server/floors/BasicPriceFloorResolver.java [hunk 1]
```diff
@@ -518,6 +544,25 @@ private static Price roundPrice(Price price) {
...
+
+        final BigDecimal floorMinValue = convertedFloorMin.getValue();
+        final String floorMinCurrency = convertedFloorMin.getCurrency();
+
+        if (StringUtils.equals(floorCurrency, floorMinCurrency) && floorValue != null && floorMinValue != null) {
+            if (floorValue.compareTo(floorMinValue) > 0) {
+                return roundPrice(effectiveFloor);
+            } else {
+                return roundPrice(convertedFloorMin);
+            }
```
Comment (And1sS) [inline src/main/java/org/prebid/server/floors/BasicPriceFloorResolver.java:559]: Can be simplified to `return floorValue.compareTo(floorMinValue) > 0 ? roundPrice(effectiveFloor) : roundPrice(convertedFloorMin)` 

We can even insert ternary into the single call to `roundPrice` but i think it's unreadable.

**Review by And1sS — CHANGES_REQUESTED** [review]

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy [hunk 2]
```diff
@@ -397,7 +397,7 @@ class PriceFloorsCurrencySpec extends PriceFloorsBaseSpec {
 
      def "PBS should choose floorMin from imp[0].ext.prebid.floors when imp[0].ext.prebid.floors is present"() {
         given: "Default BidRequest with floorMin, floorMinCur"
-        def impExtPrebidFloorMin = PBSUtils.randomFloorValue
+        def impExtPrebidFloorMin = PBSUtils.getRandomFloorValue(FLOOR_MAX, FLOOR_MIN + FLOOR_MAX)
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy:400]: Why not `getRandomFloorValue(FLOOR_MIN,FLOOR_MAX)`?

File: src/test/groovy/org/prebid/server/functional/util/PBSUtils.groovy [hunk 3]
```diff
@@ -43,8 +44,8 @@ class PBSUtils implements ObjectMapperWrapper {
         createTempFile(data, ".json")
     }
 
-    static BigDecimal getRandomFloorValue() {
-        roundDecimal(getRandomDecimal(FLOOR_MIN, 2), 2)
+    static BigDecimal getRandomFloorValue(float floorMin = FLOOR_MIN, float floorMax = FLOOR_MAX) {
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/util/PBSUtils.groovy:47]: If this method is only used in descendants of the `src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsBaseSpec.groovy`, then move it there. It's strange when you reference external constants from base test class in utility class.

**Review by Net-burst — CHANGES_REQUESTED** [review]

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy (hunk 2 above)
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy:400]: Need be confident that `impExtPrebidFloorMin` will be biggest than another `PBSUtils.randomFloorValue`

File: src/test/groovy/org/prebid/server/functional/util/PBSUtils.groovy (hunk 3 above)
Comment (Net-burst) [reply src/test/groovy/org/prebid/server/functional/util/PBSUtils.groovy:47]: Ugh, ok. I also see that we are using that method in a model too. Will need to think how to refactor that.

**Review by Net-burst — APPROVED** [review]
LGTM

**Review by And1sS — APPROVED** [review]

**Pull request merged**